tinygrad/shape/view.py

```diff
@@ -20,19 +20,29 @@ class View:
   shape:Tuple[sint, ...]
   strides:Tuple[sint, ...]
   offset:sint
-  mask:Optional[Tuple[Tuple[sint, sint], ...]]
+  mask:Optional[Tuple[Tuple[int, int], ...]]
```

Comment:
Are you sure this is okay? How does cat work?

Reply:
no that was incorrect. fixed and adding mask as a source of variable. I think it's possible to prove that only check a subset of (shape, strides, offset, mask) suffices, but checking all 4 is easier to reason